cdk/lib/dotcom-components.ts

```diff
@@ -1,79 +1,93 @@
-import { ComparisonOperator, Metric, TreatMissingData } from '@aws-cdk/aws-cloudwatch';
-import type { Policy } from '@aws-cdk/aws-iam';
-import type { App } from '@aws-cdk/core';
-import { Duration } from '@aws-cdk/core';
-import { AccessScope, GuEc2App } from '@guardian/cdk';
-import { Stage } from '@guardian/cdk/lib/constants/stage';
+import { GuEc2App } from '@guardian/cdk';
+import { AccessScope } from '@guardian/cdk/lib/constants';
 import { GuAlarm } from '@guardian/cdk/lib/constructs/cloudwatch';
 import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
 import {
-    GuDistributionBucketParameter,
-    GuStack,
-    GuStringParameter,
+	GuDistributionBucketParameter,
+	GuStack,
+	GuStringParameter,
 } from '@guardian/cdk/lib/constructs/core';
 import {
-    GuDynamoDBReadPolicy,
-    GuGetS3ObjectsPolicy,
-    GuPutCloudwatchMetricsPolicy,
+	GuDynamoDBReadPolicy,
+	GuGetS3ObjectsPolicy,
+	GuPutCloudwatchMetricsPolicy,
 } from '@guardian/cdk/lib/constructs/iam';
+import type { GuAsgCapacity } from '@guardian/cdk/lib/types';
+import type { App } from 'aws-cdk-lib';
+import { Duration } from 'aws-cdk-lib';
+import {
+	ComparisonOperator,
+	Metric,
+	TreatMissingData,
+} from 'aws-cdk-lib/aws-cloudwatch';
+import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';
+import type { Policy } from 'aws-cdk-lib/aws-iam';
+
+interface DotcomComponentsProps extends GuStackProps {
+	domainName: string;
+}
 
 export class DotcomComponents extends GuStack {
-    constructor(scope: App, id: string, props: GuStackProps) {
-        super(scope, id, props);
-
-        const appName = 'dotcom-components';
-        const baseUrl = new GuStringParameter(this, 'BaseUrl', {
-            description: 'Base URL of the service.',
-        });
-        const elkStream = new GuStringParameter(this, 'ELKStream', {
-            description: 'Name of the Kinesis stream used to send logs to the central ELK stack.',
-        });
-
-        // Cloudwatch alarms
-        const snsTopicName = 'reader-revenue-dev';
-        const namespace = `support-${appName}-${this.stage}`;
-
-        new GuAlarm(this, 'SuperModeAlarm', {
-            alarmName: `support-${appName}: Super Mode error - ${this.stage}`,
-            alarmDescription: 'Error fetching Epic Super Mode data from Dynamodb',
-            snsTopicName,
-            metric: new Metric({
-                metricName: 'super-mode-error',
-                namespace,
-                period: Duration.minutes(60),
-            }),
-            threshold: 1,
-            evaluationPeriods: 1,
-            comparisonOperator: ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
-            statistic: 'sum',
```

Comment:
Just want to confirm that's it's okay to lose the `statistic: 'sum'` bit here. Is it the default?

Reply:
good point - and actually it now goes under the `metric` object, I'll update